### ISSUE TITLE
Support for @Tools with Gemini?

### ISSUE THREAD

**Issue by waloeen** [quarkiverse/quarkus-langchain4j#1126]
I tried using `@Tools` with `quarkus.langchain4j.vertexai.gemini.chat-model.chat-model.model-id=gemini-pro`, but got the error message `Tools are currently not supported by this model`.

It seems like Gemini does support Tools: https://docs.langchain4j.dev/integrations/language-models
Also: https://github.com/quarkiverse/quarkus-langchain4j/pull/534

Are there any plans on implementing support for it?

**Comment by sberyozkin** [comment]
@geoand Yes, I am, as we'd like to show tools with Gemini :-), can you please link me to the code showing how it is done for one of the other models ?

**Comment by geoand** [comment]
Cool!

See [this](https://github.com/quarkiverse/quarkus-langchain4j/blob/cb6a4da0e64da85b816b4fc3cd110ca0595f6a15/model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaChatLanguageModel.java#L65) and [this](https://github.com/quarkiverse/quarkus-langchain4j/blob/2560cccf0cb12d629d36d18f0a0559320db1e231/model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaStreamingChatLanguageModel.java#L65)

**Comment by sberyozkin** [comment]
Thanks @geoand

**Comment by waloeen** [comment]
Thanks!
Looking forward to try this.

**Comment by sberyozkin** [comment]
Hi @waloeen @geoand I'll look at it asap, getting a bit stressed in the Quarkus space, but I'm on it, thanks

**Comment by geoand** [comment]
No rush, thanks @sberyozkin!
